projects/packages/connection/src/webhooks/class-authorize-redirect.php

```diff
@@ -97,44 +98,57 @@ function ( $domains ) {
 	}
 
 	/**
-	 * Create the Jetpack authorization URL. Copied from Jetpack class.
+	 * Create the Jetpack authorization URL.
+	 *
+	 * @since $$next-version$$ Added optional $from and $raw parameters.
 	 *
 	 * @param bool|string $redirect URL to redirect to.
+	 * @param bool|string $from     If not false, adds 'from=$from' param to the connect URL.
+	 * @param bool        $raw If true, URL will not be escaped.
 	 *
 	 * @todo Update default value for redirect since the called function expects a string.
 	 *
 	 * @return mixed|void
 	 */
-	public function build_authorize_url( $redirect = false ) {
+	public function build_authorize_url( $redirect = false, $from = false, $raw = false ) {
 
 		add_filter( 'jetpack_connect_request_body', array( __CLASS__, 'filter_connect_request_body' ) );
 		add_filter( 'jetpack_connect_redirect_url', array( __CLASS__, 'filter_connect_redirect_url' ) );
 
-		$url = $this->connection->get_authorization_url( wp_get_current_user(), $redirect );
+		$url = $this->connection->get_authorization_url( wp_get_current_user(), $redirect, $from, $raw );
 
 		remove_filter( 'jetpack_connect_request_body', array( __CLASS__, 'filter_connect_request_body' ) );
 		remove_filter( 'jetpack_connect_redirect_url', array( __CLASS__, 'filter_connect_redirect_url' ) );
 
-		/** This filter is documented in plugins/jetpack/class-jetpack.php */
-		return apply_filters( 'jetpack_build_authorize_url', $url );
+		/**
+		 * Filter the URL used when authorizing a user to a WordPress.com account.
+		 *
+		 * @since jetpack-8.9.0
+		 * @since $$next-version$$ Added $raw parameter.
+		 *
+		 * @param string $url Connection URL.
+		 * @param bool   $raw If true, URL will not be escaped.
+		 */
+		return apply_filters( 'jetpack_build_authorize_url', $url, $raw );
 	}
 
 	/**
 	 * Filters the redirection URL that is used for connect requests. The redirect
-	 * URL should return the user back to the Jetpack console.
-	 * Copied from Jetpack class.
+	 * URL should return the user back to the My Jetpack page.
 	 *
-	 * @param String $redirect the default redirect URL used by the package.
-	 * @return String the modified URL.
+	 * @param string $redirect the default redirect URL used by the package.
+	 * @return string the modified URL.
 	 */
 	public static function filter_connect_redirect_url( $redirect ) {
-		$jetpack_admin_page = esc_url_raw( admin_url( 'admin.php?page=jetpack' ) );
+		$jetpack_admin_page = esc_url_raw( admin_url( 'admin.php?page=my-jetpack' ) );
```

Comment:
This is a change in functionality from the old behavior, since we want to redirect to a page that we're sure will exist in a plugin that may use SSO, but may not use the Jetpack plugin.